### DISCUSSION TITLE
Creation portal - Review flow UX issue

### DISCUSSION THREAD

**Discussion by kiranharidas187** [Sunbird-inQuiry/Community#8] [General Discussions]
@krgauraw @pallakartheekreddy  @Krishnaj20  Currently we are facing the following UX issues

1. Currently when the Reviewer **Request changes**  a modal opens. The Reviewer will have to enter all the changes in one shot and then click on the **Submit Review** button. There is no option for the user to save the comment so that he can go through other content and update the review comment. As a real user, it's very difficult to note down all the questions in a paper and enter them in modal in one shot.
2. As a content creator, I have a button called View Comments where I can see all the changes requested by the Reviewer. But when I click on the same button on the question editor page the modal is coming as blank.

Suggested solution respectively

1. If there are APIs for saving and updating the comment, we can introduce a Save button on the Request change modal, so that the comments can be saved to the backend, and onload of the modal we can get the latest comment from the backend. Please let us know if these APIs are already available. If not available please let us know about the effort to implement it.
2. Can we show the Reviewer comment t fo the question details page also?

Please let us know your feedback on this

cc- @aks30

**Comment by Krishnaj20** [comment]
@kiranharidas187 - One of the large scale adopters of Sunbird Knowlg & inQuiry in the Indian Edtech domain has curated hundreds of thousands of assets with the existing workflow without any issues. So, it will be useful, if you can provide more details about this use case and share specifics about any experiments that were done to understand the need. Also, this will help you take a judicious call whether this capability is a 'must have' or a 'nice to have' one, and accordingly prioritize it. 
In the proposed approach, request you to think about workflows for multiple scenarios. For ex., if multiple reviewers are reviewing the asset and have added their comments, and the asset is rejected by a specific reviewer then whose comments should be shown to the creator? If you can tag the PRD to this thread, I can go thru these scenarios and add my comments. 
I'd strongly recommend to finalize the functional workflows first before starting the tech design and implementation discussions.

@pallakartheekreddy @krgauraw @alok-os

**Comment by pramodkvarma** [comment]
In general, commenting within the context is always easier for reviewers than one generic place all comments are written down. Difficult for reviewer to remember comments, make references to context of that comment, etc. Difference between Word/GoogleDoc in place comments (easy) vs sending comments via email after reading a PDF (painful).

**Comment by Prateek-slokam** [comment]
@Krishnaj20 
The use case is mostly about bringing ease
1. To the reviewer for putting up their comments for the changes that they want to suggest.
         The observation and survey form are different than the till date used question sets as each question is trying to capture the opinions, requirements or present availability for the user and thus, each question and the entire form has a subjectivity with a derived context of the location, purpose and persona for which it is designed. The reviewer needs to follow onto this context while reviewing and further specifically need to give comments according to the matter of each question, question set (like sections or criteria) within the entire question set of observation and survey. To do this in a single go would require the reviewers to lay down their list of comments separately first and mark them in platform with the matter to which it points.
         Eg. We have seen this reviewing behaviour of contextual comments to the questions and question sets in the observation and survey in works in the current design of template based creation. The comments if are specific to question or say a particular score or hint/tip in the question, would not make sense without the comment pointing to that matter.
2. To the creator to easily and clearly understand the suggestions that have been recommended with the context of substance the change have been recommended in.
        As within the above example the creator too needs to have an idea of the particular point of reference of a suggestion. As each question has a different information to capture with no 'Book designed format' for the questions and question sets, it becomes tricky for creator otherwise to understand via a single comment all the changes that are to be made.
3. In saving of the reviewer's work and navigation by content creator through the form
         The length of the observation forms are huge, especially like in the case of an observation with rubric where a general form might end up having about 80-100 questions. In such a case to go through the entire forms in a single go would be difficult and so for reviewer to be able to put up comments at each individual page of creator and rather than in a single go would make a huge difference in the experience and efficiency. With the same idea the creator would also feel lost with a single comment box for so many questions. 


The ultimate Plan for a review flow, is in line with this above mentioned use case and I think to what @pramodkvarma sir is also suggesting. You can look up the Jira ticket - [[28057](https://project-sunbird.atlassian.net/browse/SB-28057)]. The UI is also attached in the same. 
This is the [PRD](https://project-sunbird.atlassian.net/wiki/spaces/SingleSource/pages/2822963275/Observation+Survey+creation+portal+PRD#Review-Flow) for your reference.  

However, we have upon discussions with the engineering team also realised that this is a larger effort which will require longer time and so as the not delay the release because of this single flow we are looking at it into two phases now. The intermediatory solution (or phase 1) is what Kiran mentioned in the discussion where we are looking to at least make the reviewer part easier by providing them option of continuously being able to add and save comments and the creator having the access to the comment dialog box at all places. In phase 2 (which would be after the first release) we are targeting for achieving the design mentioned in the PRD to bring ease and contextuality to the entire review flow.

**Reply by Krishnaj20** [reply]
@Prateek-slokam - thanks for providing the context. This is useful. I'd recommend you to change the scope of the story as per the phase1 implementation scope. Phase 2 can be added to the backlog. Apologies, if this is already taken care of.
@kiranharidas187 - Since this is a new feature, I'm assuming the tech design was already run with @vrayulu @maheshkumargangula @pallakartheekreddy ? If not, I'd recommend you to share the tech design document for feedback/clarifications.

**Comment by Prateek-slokam** [comment]
@Krishnaj20 Sorry for the delay in response, The stories for the first phase review flow enhancements are put up - https://project-sunbird.atlassian.net/browse/SB-29586 and https://project-sunbird.atlassian.net/browse/SB-29587. @kiranharidas187 Can you please take it forward as Krishna is suggesting.

**Reply by Krishnaj20** [reply]
@Prateek-slokam @aks30 - In the last meeting b/w Ekstep & SL teams, @maheshkumargangula has volunteered to help SL team come up with the initial design and the implementation design for this contribution. I hope you folks have made progress on that front. 
In the spirit of Open source contributions, I request you to follow the below process:
a) Once, you with the help @maheshkumargangula, have the initial design ready, pls share it in this thread for the review of @pallakartheekreddy @vrayulu & the community. We'd recommend you to proceed with the implementation after this review.
b) Are the PRD & JIRA stories shared in this thread ready for review? Pls let me know - I'll review and add my comments. Also I'd request you to include wireframes for all the scenarios. The existing design in the story has only one screen.

Pls do reach out to us if you have any queries.

@alok-os @madhucr

**Comment by kiranharidas187** [comment]
MOM of call held on 12th May 

- As the suggestion made above, we will be implementing a save button in the Request change modal.  Story ref - https://project-sunbird.atlassian.net/browse/SB-29587
- Complete comments flow will be implemented in the next phase
- Mahesh will talk to DC and let us know if this approach is fine by  13th May EOD

cc- @Krishnaj20 @pallakartheekreddy @aks30 @maheshkumargangula

**Reply by maheshkumargangula** [reply]
@kiranharidas187 @aks30 - Let us have a PRD explaining the capability that we are planning to enable here and have a high-level design to support the requirement. After that, we will be able to decide on whether the shortcut implementation whatever is proposed by you in our previous call is valid or not. 

Also, along with the shortcuts, we should plan and define the required changes to implement the final version of this feature and any changes and data migration involved to deploy the final version of this feature.
Please define these and schedule a call with DC and other required people for review.
Cc: @Krishnaj20 @pallakartheekreddy @vrayulu

**Reply by aks30** [reply]
@maheshkumargangula During the call on 28th April, we in the interest of time agreed on the Phase 1 approach and decided to subsequently come up with the design and implementation as per the PRD @Prateek-slokam has shared as that entails a whole new feature development from scratch and would require much time to be spent on coming up with the design of the same. Can we for now go ahead without that being a blocker?

cc - @alok-os, @vijiurs, @Krishnaj20

**Reply by vijiurs** [reply]
@maheshkumargangula ,   The Phase 1 enhancement enables a reviewer to keep saving the review feedback in draft mode as he/she reviews the question set.   This feature would be required in the final solution too. Given this,  I think we should go ahead with the Phase 1 design and implementation and not wait for the design of the final solution.  
@aks30 ,  Please share the design and implementation details with @maheshkumargangula and team.  
 cc- @alok-os, @Krishnaj20 , @madhucr.

**Comment by kiranharidas187** [comment]
@maheshkumargangula @pallakartheekreddy  We have created a [confluence doc ](https://project-sunbird.atlassian.net/l/c/htp5NV1q)for the Phase 1  approach.  Please let us know if this approach is fine.

cc- @vijiurs

**Reply by kiranharidas187** [reply]
@maheshkumargangula @pallakartheekreddy  We have updated the  UI mock screens in the doc as mentioned above. Please check and let us know your feedback

**Reply by maheshkumargangula** [reply]
@kiranharidas187 - We should not use the update API to save comments. Below is the feedback for the approach.

- Define a separate API to save the comments. The API Spec should be generic to support other object types from Sunbird BB.
- Using Update API, we should not be able to modify the comments. Let us ensure this with our design.
- The comments data should not sync to ElasticSearch along with other metadata.

Please consider the above changes in the Tech Design and let us know.
Cc: @pallakartheekreddy @Krishnaj20

**Reply by joffinjoy** [reply]
@maheshkumargangula Here is the approach we came up with and the questions regarding the same. We will update the confluence doc once we have some clarity.

Approach:
We will have a new field called "reviewerComments" in cassandra for questionSet table - "hierarchy_store.questionset_hierarchy".
We will write a new questionset API "addReviewComment" for updating this "reviewerComments", the saved comment will be read using the current QuestionSet Read API.

Questions:
Is this approach fine?
Any suggestion on how the same workflow can be used for other object types?

cc: @aks30 @kiranharidas187 @vijiurs

**Reply by vijiurs** [reply]
@maheshkumargangula,  Please review and share your comments on the approach.

**Reply by maheshkumargangula** [reply]
@joffinjoy - The approach looks fine but, please define the API specification to save the comments along with the approach details and share it with DC for approval. 
Cc: @vijiurs @pallakartheekreddy @Krishnaj20

**Reply by joffinjoy** [reply]
@maheshkumargangula What's your suggestion regarding the second question? The one about other object types. We have not been given any pointers on how to accommodate them. Please clarify.
cc: @aks30 @vijiurs

**Reply by joffinjoy** [reply]
@maheshkumargangula I have still not been able to figure out where to put the implementation that supports other object types. Or the expected workflow to support them? Do I write an implementation that accepts a node or request(which can be or related to any objectType) in Assessment Manager or HierarchyManager or some place else?
cc: @aks30 @vijiurs

**Reply by joffinjoy** [reply]
@maheshkumargangula I have updated the confluence doc with the new proposed implementation and assumptions. Please check it out and let me know what modification should be made. 
https://project-sunbird.atlassian.net/l/c/ktmtdC1b
cc: @aks30 @vijiurs

**Reply by maheshkumargangula** [reply]
@joffinjoy - Please define the OpenAPI Specification for the "Add Comment API" and include the details in the design wiki. Please use Swagger to define the API Specification. 
Please connect with @krgauraw  or @pallakartheekreddy if you need help in understanding this.

@vijiurs

**Reply by kameshbhr** [reply]
I have put in a couple of comments in the confluence doc.
One point I would like to be captured explicitly in the design: Does the new implementation impact existing systems in anyway?
- Any other front-end apps that use the Question/Question Set API? 
- Any existing question/question set content that was created earlier?

**Reply by joffinjoy** [reply]
@maheshkumargangula I have added the API specification file to the confluence doc. https://project-sunbird.atlassian.net/l/c/FnQvHVHb

@kameshbhr Replied to your comments in the linked document. But reiterating for future reference here: Since this is a new API which simply adds an external data field "reviewerComments" to Cassandra, it doesn't affect existing system. All previously created question/questionsets stays unaffected by the usage of this field. 
cc: @aks30 @vijiurs

**Reply by joffinjoy** [reply]
@maheshkumargangula Any update on this? 

@pradeepkumarcm96 Follow this discussion for reviewer-flow.
cc: @aks30

**Reply by maheshkumargangula** [reply]
@joffinjoy - The API request should capture the minimum required context information. The comment is for the QuestionSet itself or for one of the children of QuestionSet? 

What is the data model or data type to save this data in the Cassandra table?  We should capture the context information like timestamp, the object identifier for which the comment is and the userId who is giving the comment etc,.

Also, the existing implementation doesn't understand the new column(which we are introducing for comments). You might need to make some code changes to return the comments information in the read hierarchy response. 

Do we need to maintain the comments based on the object version?

Please check for all the APIs and scenarios to understand the impacted areas and propose the tech design based on it.

Cc: @aks30 @vijiurs @surendrasinghs @pallakartheekreddy

**Reply by aks30** [reply]
@maheshkumargangula Please find my response inline:-

> The API request should capture the minimum required context information. The comment is for QuestionSet itself or for one of the children of QuestionSet?

As part of Phase 1, we are not changing the overall construct/design that currently exists for reject comments i.e. to capture 1 single reject comment for the entire question set. We are only introducing an intermediate "Save Comment" option so that the user can keep on updating and saving his comments without having to make a note outside and then submit them at once which is what he has to do right now. We have given the implementation design bearing in mind the only change we are going to make is to not save the intermediate comments draft (when the reviewer clicks on save) in Neo4j as it subsequently gets synced to ES and that is something that we should avoid as it can happen that a comment gets updated several times during the review and we should not be updating the ES record so often. Since at present there is no context saved for the reject comments for any object type we do not plan to enhance the same and we will look at in its entirety during the Phase 2 design.


> What is the data model or data type to save this data in the Cassandra table? We should capture the context information like timestamp, the object identifier for which the comment is, and the userId who is giving the comment, etc.

Again we are going with the same data structure that currently exists for reject comments i.e. **string**. Phase 1 doesn't require us to save any context. We will be considering that when we design for Phase 2 as that will have other information too which has to be part of the context.

> Also, the existing implementation doesn't understand the new column(which we are introducing for comments). You might need to make some code changes to return the comments information in the read hierarchy response.

From what we understand Cassandra data or external properties of question sets are returned only in questionSet read API response (given the parameter is explicitly specified in the request) and not in the hierarchy read API. Let us know if that understanding is wrong. Also, should the review comments be part of both the APIs or only the hierarchy read API in that case?


> Do we need to maintain the comments based on the object version?

No, once the question set is published, for the next iteration of edit and publish it doesn't make sense to show the previous review comments since it's a fresh review every time.

> Please check for all the APIs and scenarios to understand the impacted areas and propose the tech design based on it.

I am not sure I understand - "Please check for all the APIs and scenarios to understand the impacted areas and propose the tech design based on it." Exactly which APIs and scenarios we should be checking for impact? As in this case, we are making a new API and adding a new key to the existing API response. We are not modifying any existing response value or key. Also, we have shared the design - [here](https://project-sunbird.atlassian.net/l/c/FnQvHVHb) , do let us know if anything further we should cover that.


cc: @joffinjoy @vijiurs @kiranharidas187 @Prateek-slokam @pradeepkumarcm96, @aishwaryashikshalokam, @MeghnaSL, @githubpiyush

**Reply by Prateek-slokam** [reply]
@maheshkumargangula Can you please clear the doubts at the earliest. This really is a high priority item and these delays are putting the entire implementation at risk. As per our prior conversations this was set to be rolled out in the 5.0 release which is just couple of weeks into closing.

**Reply by maheshkumargangula** [reply]
@aks30 @joffinjoy  - As we discussed yesterday the comments should have minimum required context information. This will help us to migrate the existing data with ease when we enhance this capability in Phase 2.
Also, please remove the `reviewer` keyword in the request body of the API specification. Just use `comment`.

Cc: @Prateek-slokam @vijiurs

**Reply by aks30** [reply]
@maheshkumargangula Contrary to what you mentioned, what I recall from our discussion yesterday is that we will not do so right now because if we do then ideally we should be capturing the same against reject comments as well as for the approving flow. And we don't need backward compatibility on this for now as this is a stop-gap solution and these comments are not of much use once the question set is published.

cc - @joffinjoy , @vijiurs @Prateek-slokam

**Reply by vrayulu** [reply]
@aks30 - if an offline discussion happened for a thread, please post the minutes of the meeting on the thread so that others will also have the complete context.

Coming to the design: contextual comments should be the way forward. the context could be main object itself, or one of the child objects (in case of collections, question sets) or a page/slide/moment/section of an object (e.g. specific page of a PDF, slide in an ecml content, timestamp in a video, option in a MCQ question, or a specific metadata field like name, description,  etc).

Keeping the above in mind, we can start with comments enabled only for the primary object (i.e. the collection, content, question set or the question). Below are my comments on the current design and a few queries on the feature:
- the API should not be specific for question set, it should be applicable for any object type.
- there should be APIs to save and retrieve comments for a specific object and/or context.
- audit information about the comment must be captured, e.g. who, when, etc.
- it is must to tag comments with the version of the object. you may not have the need to show comments of older versions anywhere. But it is required to differentiate the comments for latest version vs previous versions in the database (and for other purposes in future).
- proper telemetry should be generated for comments
- database design is not clear in the design wiki. please add more details.
- can multiple reviewers add comments to the same object while it is in review? i think it is possible, and if so, the design should support it.
- how does the UI look like for the creator to view the comments?

**Reply by pramodkvarma** [reply]
Absolutely agree with @vrayulu here. Also why can't we use discussions APIs to attach comments feature on various content objects? Why do we need a new API at all?

**Reply by vrayulu** [reply]
Thats a great idea @pramodkvarma . Discussion forums API already have the concept of context and linking the context to a forum. 

@aks30 - please check discussion forum APIs. we should be able to create a forum for each object's review comments and use tags on topics & posts for additional context information.

**Reply by pramodkvarma** [reply]
Absolutely! Please don't create more APIs than necessary. It gives full capability to do threaded replies etc and can be attached to any context. Glad I intercepted this thread 😄

**Reply by joffinjoy** [reply]
@maheshkumargangula @vrayulu @pramodkvarma 
I have looked into discussion APIs in Sunbird Lern BB and have updated the documented with the alternative implementation. Please check them out and verify if the suggested mapping between questionset/question and category/topic/posts are valid or not. 

Doc Link: https://project-sunbird.atlassian.net/wiki/spaces/DEB/pages/3159916549/Creation+Portal+Reviewer+flow+-+Save+Intermediate+Review+Comment#Alternative-Implementation

Reiterating the same here: 

> As per our current understanding, we can create a new category called “Reviewer Comments” in nodeBB, and then obtain the category Id to be used in further requests. As such, **a topic can be created for each questionSet** that is under review, and reviewer can add as many comments as they want as "**posts**" under that particular **topic**. 

Is this assumption and mapping correct?
cc: @aks30 @vijiurs

**Reply by pramodkvarma** [reply]
Looks correct to me. @vrayulu ?

**Reply by vrayulu** [reply]
@joffinjoy - the approach looks fine at a high level. I think we should figure out a few next level details also before getting into the implementation:

- how is the topic to questionSet association stored? - as tags on the topic or as metadata in the questionSet? I think the hierarchy in discussion forums is category - forum - topic - posts. it might make more sense if we have one forum per questionSet instead of one topic per questionSet and we can use one topic per questionSet version.
- are there any tags that we want to set on topics and posts for this use case?
- there are APIs to add users to a forum and privileges assigned to the users for the forum. do we need to add creators & reviewers as users to the forum? and what privileges would they have?
- how do the lifecycle of a topic correlate with the questionSet lifecycle? e.g.: will topic get auto-closed when the questionSet is published?

**Reply by joffinjoy** [reply]
> how is the topic to questionSet association stored? - as tags on the topic or as metadata in the questionSet?

We would like to store the discussion ids in questionSet as metadata since that seems like the most straight-forward option given that we are fetching the questionSet first and showing it to reviewers and creator. So having the ids with questionSet would make it easy to fetch the comments from discussion apis. Additionally we couldn't figure out how to search/retrieve posts using custom tags (questionSet Ids in this case). 

> are there any tags that we want to set on topics and posts for this use case?

According to our present requirements we couldn't find the the need for storing any additional information as tags comments and replies. In future, maybe we can use them if the need arise. 

> we can use one topic per questionSet version

When we talk about questionSet versions, are we talking about questionSet versioning between a published questionset and modification of that particular questionSet resulting in a new version which is stored as an **image node**? 

I am not sure we need a version history in that case, since the second version (image node) is generated only after the questionSet is published **once** and then modified. Meaning, A reviewer already found the original questionSet to be good enough for publish. Review of a image node doesn't feel related to the published questionSet and its review process.

Or did you mean versioning as in reviewer reviews a questionset, creator changes the questionset according to the comments and then sends it again for the review. Now reviewer should be able to see the previous comments from previous review rounds. Is that the point of review comment versions? Because calling it "questionset version" is confusing, since creator is making changes to the original questionset no matter how many times that questionset has been reviewed (no versioning of questionsets).

**Reply by pramodkvarma** [reply]
It's not a good idea to store actual discuss IDs within question set metadata. Simply create a topic ID with same ID as question set + version (something unique and consistently derivable). That way you never have to store anything as metadata. Simply call discussion API with topic ID (derived from question set + version or whatever is unique) and retrieve all discussions under that topic. @joffinjoy @vrayulu

**Reply by aks30** [reply]
@pramodkvarma that would be great if we could do that, we did check the APIs of the Discussion forum but couldn't find a way to set our own custom identifier against posts and topics and retrieve using the same.

**Reply by pramodkvarma** [reply]
Basically you should create a topic dynamically for every unique content item with IDs that are derived. If discussion APIs do not allow creation of topics with custom ID/name (I think it does). Btw topic name also need not be human readable since these are internal topics that only shows up in embedded form within content. So either ID or name can be constructed in a unique fashion using content ID.

If none of the above works, I suppose you will have to store topic ID as part of content metadata (not all discussion IDs, just parent topic ID).

@vrayulu can you please help?

**Reply by vrayulu** [reply]
@joffinjoy @aks30 - here is an option for using discussion forums to manage review comments on assets (content, collection, question, questionSet, etc):

- create a discussions category "Reviewer Comments"
- create one forum per asset dynamically when the asset is created (or submitted for review for the first time). You can provide the object type (e.g.: QuestionSet) and object identifier (e.g.: question set identifier) as input to the createForum API. And there is an API fetch the forum created for a specific object.
- within the forum, create multiple topics - one per one version of the object (@joffinjoy - each time an object is published, the version number of the object is updated in KP. we need to use this version number to differentiate the review comments given in each cycle of review-publish). And use the tags capability to add the version number as a tag on the topic. 
- within each topic, there could be multiple posts for comments by the reviewer and replies by the creator.  

Hope this helps... however, please validate this design for multiple scenarios and see if it is able to fulfil all of them.

**Reply by pramodkvarma** [reply]
Yes. Thanks @vrayulu

**Reply by joffinjoy** [reply]
@vrayulu 

> create one forum per asset dynamically when the asset is created (or submitted for review for the first time)

We are confused about the concept of forums here. I mean, from the documentation it seems like the exact hierarchy is **Category -> Sub-Category -> Topics -> Posts**. This was confirmed by @VenkatKadiveti in the discussion post I opened recently. Question #1 in https://github.com/orgs/Sunbird-Lern/discussions/27. Are you using forums and Sub-Categories interchangeably? 

Going through Forum APIs listed at: https://lern.sunbird.org/learn/product-and-developer-guide/discussion-forum/api-documentation/forum-apis isn't helping me clear this confusion for myself. Category (or sub-category) feels synonymous to forums. 

> You can provide the object type (e.g.: QuestionSet) and object identifier (e.g.: question set identifier) as **input** to the createForum API. 

Does **input** here mean the tags or do you want it to be part of the forum id like @pramodkvarma suggested for posts (Using **questionSet Id** as the forum name)? My query #3 in the linked post was regarding fetching posts/topics by their tags. Venkat shared the curl that will do that. Does the same apply to forums as well? Because we couldn't find an API to create a forum and pass in object ids in the 'API listing page'. Can you please clarify this? 

> within the forum, create multiple topics - one per one version of the object

Okey! Understood exactly what versioning you meant. But in that case, a second version happens when a questionSet that is already published undergoes a modification, leading to the creation of an image node(at this point, the original questionset is untouched). And this image node is the one that is getting reviewed. But since we are not going to show previous comments from earlier review-publish cycle, is this necessary?

But instead if we are considering versioning between creator and reviewer, Like creator sends a questionset for review and reviewer reviews it and sends it back, this becomes **version one**. Then the creator modifies the QS according to the review comments and sends it again for review, this time it becomes **version 2 of reviewer comments**. This isn't any way related to publish process at all. Basically each **creation-review** or **review-modify-review** becomes a version of its own. Wouldn't this be a better approach as far as versioning is concerned?

cc: @VenkatKadiveti

**Reply by vrayulu** [reply]
@aks30 @joffinjoy - if you need more help in doing this design, lets get into a call and you will be able to understand it better. Please write to me on my email and we can identify a time.

P.S.: the summary of the call should be added back to this thread post the call.

**Comment by aks30** [comment]
We (@vrayulu ,  @Prateek-slokam , @vijiurs , @kiranharidas187 ) have had a meeting and confirmed taking this requirement in 2 phases.

Phase 1 - Allow the reviewer to save comments as a draft before he clicks on the final publish/reject.
Phase 2 - Bring in discussion forum to enable Google Docs kind of comment for questionSet and collections.

We have put out the design document as per our discussion for Phase 1 here - https://project-sunbird.atlassian.net/l/cp/my7WTrV7

@vrayulu request you to check and share if any comments or changes on the same.

**Reply by kameshbhr** [reply]
I have a few questions on the design that I have added in the confluence page.

**Reply by aks30** [reply]
Have added responses for the same

**Reply by kameshbhr** [reply]
I have further added my comments. Would like @vrayulu to go through it. If required we can plan a quick call to close on it.
